### PR TITLE
Add Contracts module in graphql client

### PR DESCRIPTION
**Pull request by Basem1166**
[ticket link](https://github.com/codescalers/tfgrid-sdk-dart/issues/32)

### CODE REVIEW

**Comment by Basem1166** [comment]
- added toJSON methods and refactored the parsing statements in the src code

File: packages/graphql_client/lib/src/contracts.dart [hunk 1]
```diff
@@ -1,14 +1,232 @@
 import 'dart:math';
```
Comment (AlaaElattar) [inline packages/graphql_client/lib/src/contracts.dart:1]: Please remove this unused import.

**Review by AlaaElattar — COMMENTED** [review]
- In `contract_state` file, please change the enum to be consistent with the ones u created before.

**Comment by AlaaElattar** [comment]
- Please check this error.

image

File: packages/graphql_client/lib/models/contracts.dart [hunk 2]
```diff
@@ -23,95 +30,319 @@ DiscountLevel parseDiscountLevel(String level) {
...
+    super.state,
+    super.createdAt,
+    super.solutionProviderID,
+    this.name,
   });
+
+  factory GqlNameContract.fromJson(Map<String, dynamic> json) {
+    return GqlNameContract(
+      id: json['id'] ?? '',
+      gridVersion: json['gridVersion'] ?? '',
```
Comment (AlaaElattar) [inline packages/graphql_client/lib/models/contracts.dart:128]: `gridVersion` is int not string, and this will cause the function to crash.

**Review by AlaaElattar — APPROVED** [review]

**Pull request merged**